modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java

```diff
@@ -132,33 +136,206 @@ private void _addObjectEntries(Integer numberOfEntries) throws Exception {
 				},
 				ObjectDefinitionConstants.SCOPE_COMPANY);
 		}
+
+		try (Closeable closeable = new PerformanceTimer(
+				GetterUtil.getInteger(
+					_properties.getProperty("object.entries.get.max.time")),
+				"Get all the Object Entries by ObjectEntryLocalService with " +
+					"CustomObjectDefinitionId:" +
+						_customObjectDefinition.getObjectDefinitionId())) {
+
+			_objectEntryLocalService.getObjectEntries(
+				0, _customObjectDefinition.getObjectDefinitionId(),
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+		}
+	}
+
+	@Test
+	public void testHttpPostAndDeleteObjectEntries() throws Throwable {
+		TransactionConfig.Builder transactionConfigBuilder =
+			new TransactionConfig.Builder();
+
+		_company = TransactionInvokerUtil.invoke(
+			transactionConfigBuilder.setPropagation(
+				Propagation.REQUIRED
+			).setRollbackForClasses(
+				Exception.class
+			).build(),
+			() -> {
+				Company company = CompanyLocalServiceUtil.addCompany(
+					null, _VIRTUAL_HOST_NAME, _VIRTUAL_HOST_NAME,
+					_VIRTUAL_HOST_NAME, 0, true, true, null, null, null, null,
+					null, null);
+
+				PortalInstances.initCompany(company);
+
+				return company;
+			});
+
+		try {
+			_invokeHttp(null, HttpInvoker.HttpMethod.GET, _getPath(""));
```

Comment:
@slnn We should move this check to the very beginning of this test method.

Reply:
@slnn Yes that's what I meant. Creating a new company is costly. If the hosts is not properly configured, there's no point to create the company.